src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -307,14 +352,17 @@ private static bool HasReferenceToAssembly(Project project, string assemblyName,
             if (metadataId is null)
                 continue;
 
-            if (!s_metadataIdToAssemblyName.TryGetValue(metadataId, out var name))
+            using (await s_metadataIdToAssemblyNameGate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
             {
-                uncomputedReferences.Add((peReference, metadataId));
-                continue;
-            }
+                if (!s_metadataIdToAssemblyName.TryGetValue(metadataId, out var name))
```

Comment:
>if (!s_metadataIdToAssemblyName.TryGetValue(metadataId, out var name)) [](http://example.com/codeflow?start=16&length=70)

might be nice to match below:

if (s_metadataIdToAssemblyName.TryGetValue(metadataId, out var name) && name == assemblyName)
    return true;

and then move the uncomputedReferences outside the gate

Reply:
done.